chapter16/dcox/MyDate.pm

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/perl
+package MyDate;
+
+use strict;
+use warnings;
+
+use Carp qw(croak);
+
+use vars qw($AUTOLOAD);
+
+sub new { bless {}, $_[0] }
+
+sub AUTOLOAD {
+  our $AUTOLOAD;
+
+  ( my $method = $AUTOLOAD ) =~ s/.*:://s; # remove package name
+
+  if ( $method eq "date" ) {
+    return ( localtime )[3];
+  }
+  elsif ( $method eq "month" ) {
+    return ( localtime )[4] + 1;
+  }
+  elsif ( $method eq "year" ) {
+    return ( localtime )[5] + 1900;
+  }
+  else { # unknown method
+    croak "$_[0] does not know how to $method\n";
+  }
+}
```

Comment:
This works, yeah. It does mean that `date/month/year` subroutines will never actually be created on the object, but then the exercise never explicitly asked for that, so we're good. (I did a similar thing myself).